// react-responsive-pagination/lib.cjs/widthCalculator/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createWidthCalculator = void 0;
var totalWidthCalculator_1 = require("./totalWidthCalculator");
var itemWidthCalculator_1 = require("./itemWidthCalculator");
var navWidthCalculator_1 = require("./navWidthCalculator");
var pageWidthCalculator_1 = require("./pageWidthCalculator");
var numberWidthCalculator_1 = require("./numberWidthCalculator");
var paginationItem_1 = require("../paginationItem");
function createWidthCalculator(baseMetrics) {
    if (!baseMetrics) {
        return { requiredBaseMetrics: baseMetricItemsToMeasure };
    }
    return createGraph(baseMetrics);
}
exports.createWidthCalculator = createWidthCalculator;
function createGraph(baseMetrics) {
    var itemWidths = baseMetrics.itemWidths, outerFrameWidth = baseMetrics.outerFrameWidth;
    return (0, totalWidthCalculator_1.createTotalWidthCalculator)({
        getItemWidth: (0, itemWidthCalculator_1.createItemWidthCalculator)({
            getPageWidth: (0, pageWidthCalculator_1.createPageWidthCalculator)({
                getNormalPageWidth: (0, numberWidthCalculator_1.createNumberWidthCalculator)({
                    singleDigit: itemWidths.normalPageSingleDigit,
                    doubleDigit: itemWidths.normalPageDoubleDigit,
                }),
                getActivePageWidth: (0, numberWidthCalculator_1.createNumberWidthCalculator)({
                    singleDigit: itemWidths.activePageSingleDigit,
                    doubleDigit: itemWidths.activePageDoubleDigit,
                }),
            }),
            getNavWidth: (0, navWidthCalculator_1.createNavWidthCalculator)({
                previous: {
                    enabled: itemWidths.navPreviousEnabled,
                    disabled: itemWidths.navPreviousDisabled,
                },
                next: {
                    enabled: itemWidths.navNextEnabled,
                    disabled: itemWidths.navNextDisabled,
                },
            }),
            ellipsisWidth: itemWidths.ellipsis,
        }),
        outerFrameWidth: outerFrameWidth,
    });
}
var baseMetricItemsToMeasure = {
    normalPageSingleDigit: (0, paginationItem_1.createPageItem)(8, false),
    normalPageDoubleDigit: (0, paginationItem_1.createPageItem)(88, false),
    activePageSingleDigit: (0, paginationItem_1.createPageItem)(8, true),
    activePageDoubleDigit: (0, paginationItem_1.createPageItem)(88, true),
    navPreviousEnabled: (0, paginationItem_1.createNavItem)('previous', 0),
    navPreviousDisabled: (0, paginationItem_1.createNavItem)('previous'),
    navNextEnabled: (0, paginationItem_1.createNavItem)('next', 0),
    navNextDisabled: (0, paginationItem_1.createNavItem)('next'),
    ellipsis: (0, paginationItem_1.createEllipsisItem)('left'),
};
