// react-responsive-pagination/lib.cjs/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var prop_types_1 = __importDefault(require("prop-types"));
var usePaginationItems_1 = require("./hooks/usePaginationItems");
var dom_1 = require("./helpers/dom");
exports.default = (0, react_1.memo)(BootstrapPagination);
/* eslint-disable jsx-a11y/anchor-is-valid */
function BootstrapPagination(_a) {
    var current = _a.current, total = _a.total, handlePageChange = _a.onPageChange, maxWidth = _a.maxWidth, _b = _a.extraClassName, extraClassName = _b === void 0 ? 'justify-content-center' : _b, previousLabel = _a.previousLabel, nextLabel = _a.nextLabel;
    var _c = (0, usePaginationItems_1.usePaginationItems)(current, total, maxWidth, {
        previousLabel: previousLabel,
        nextLabel: nextLabel,
    }), items = _c.items, ref = _c.ref;
    if (items.length === 0)
        return null;
    var className = 'pagination';
    if (extraClassName) {
        className += ' ' + extraClassName;
    }
    return (react_1.default.createElement("ul", { className: className, ref: ref }, items.map(function (item) {
        return item.gotoPage !== undefined ? (
        // item = ClickableItem
        react_1.default.createElement("li", { key: item.key, className: "page-item".concat(item.active ? ' active' : '') },
            react_1.default.createElement("a", { className: "page-link", href: "#", onClick: (0, dom_1.preventDefault)(function () { return handlePageChange(item.gotoPage); }), "aria-label": item.a11yLabel }, getLabel(item.label, item.a11yLabel)))) : (
        // item = NonClickableItem
        react_1.default.createElement("li", { key: item.key, className: "page-item disabled", "aria-hidden": item.a11yHidden },
            react_1.default.createElement("span", { className: "page-link" }, getLabel(item.label, item.a11yLabel))));
    })));
}
BootstrapPagination.propTypes = {
    current: prop_types_1.default.number.isRequired,
    total: prop_types_1.default.number.isRequired,
    onPageChange: prop_types_1.default.func.isRequired,
    maxWidth: prop_types_1.default.number,
    extraClassName: prop_types_1.default.string,
    previousLabel: prop_types_1.default.string,
    nextLabel: prop_types_1.default.string,
};
function getLabel(label, a11yLabel) {
    return a11yLabel ? (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement("span", { "aria-hidden": "true" }, label),
        react_1.default.createElement("span", { className: "sr-only" }, a11yLabel))) : (label);
}
