// react-responsive-pagination/lib.cjs/paginationItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createEllipsisItem = exports.createPageItem = exports.createNavItem = void 0;
function createNavItem(type, gotoPage) {
    return {
        type: type,
        key: gotoPage === undefined ? type : "".concat(type, "_disabled"),
        label: type === 'previous' ? '«' : '»',
        a11yLabel: type === 'previous' ? 'Previous' : 'Next',
        gotoPage: gotoPage,
    };
}
exports.createNavItem = createNavItem;
function createPageItem(page, active) {
    return {
        type: 'page',
        key: active ? "page_".concat(page) : "active_".concat(page),
        label: page.toString(),
        a11yLabel: active ? '(current)' : undefined,
        gotoPage: page,
        active: active,
    };
}
exports.createPageItem = createPageItem;
function createEllipsisItem(position) {
    return {
        type: 'ellipsis',
        key: "ellipsis_".concat(position),
        label: '…',
        a11yHidden: true,
        gotoPage: undefined,
    };
}
exports.createEllipsisItem = createEllipsisItem;
